### ISSUE TITLE
Method overwriting / precompilation errors (Julia 1.10.2)

### ISSUE THREAD

**Issue by vram0gh** [stefftaelman/Xicor.jl#5]
Thanks for creating this package.  It looks like maybe some changes in Julia since this package was published may be triggering issues that weren't seen previously?

Please see errors in the precompilation step on Julia 1.10.2, Linux/AMD64 below.  There are also documentation warnings generated:

```
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.10.2 (2024-03-01)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using Xicor
Precompiling Xicor
        Info Given Xicor was explicitly requested, output will be shown live
WARNING: Method definition maxrank(Any) in module Xicor at /home/vram0gh/.julia/packages/Xicor/BajYB/src/rank.jl:9 overwritten on the same line (check for duplicate calls to `include`).
ERROR: Method overwriting is not permitted during Module precompilation. Use `__precompile__(false)` to opt-out of precompilation.
  ? Xicor
[ Info: Precompiling Xicor [77194b58-d7c3-46dd-b530-0e56265cadd8]
WARNING: Method definition maxrank(Any) in module Xicor at /home/vram0gh/.julia/packages/Xicor/BajYB/src/rank.jl:9 overwritten on the same line (check for duplicate calls to `include`).
ERROR: Method overwriting is not permitted during Module precompilation. Use `__precompile__(false)` to opt-out of precompilation.
[ Info: Skipping precompilation since __precompile__(false). Importing Xicor [77194b58-d7c3-46dd-b530-0e56265cadd8].
┌ Warning: Replacing docs for `Xicor.maxrank :: Tuple{Any}` in module `Xicor`
└ @ Base.Docs docs/Docs.jl:243
┌ Warning: Replacing docs for `Xicor.minrank :: Tuple{Any}` in module `Xicor`
└ @ Base.Docs docs/Docs.jl:243
┌ Warning: Replacing docs for `Xicor.randrank :: Tuple{Any}` in module `Xicor`
└ @ Base.Docs docs/Docs.jl:243
┌ Warning: Replacing docs for `Xicor.rank :: Tuple{Any}` in module `Xicor`
└ @ Base.Docs docs/Docs.jl:243
```

**Comment by juliohm** [comment]
I can reproduce this issue. Please consider a hot fix.

**Comment by stefftaelman** [comment]
Thanks for the notice, the package was in dire need of an update. This should have been fixed in v0.2.1 with basic functionality. Planning larger updates on the longer term, feel free to make issues if you have suggestions!
